ext/smarter_csv/smarter_csv.c

```diff
@@ -15,64 +15,70 @@
 static VALUE rb_parse_csv_line(VALUE self, VALUE line, VALUE col_sep, VALUE quote_char, VALUE max_size) {
   if (RB_TYPE_P(line, T_NIL) == 1) {
     return rb_ary_new();
+  }
+
+  if (RB_TYPE_P(line, T_STRING) != 1) {
+    rb_raise(rb_eTypeError, "ERROR in SmarterCSV.parse_line: line has to be a string or nil");
+  }
 
-  } else if (RB_TYPE_P(line, T_STRING) == 1) {
-    rb_encoding *encoding = rb_enc_get(line); /* get the encoding from the input line */
-    char *startP = RSTRING_PTR(line); /* may not be null terminated */
-    long line_len = RSTRING_LEN(line);
-    char *endP = startP + line_len ; /* points behind the string */
-    char *p = startP;
+  if (RSTRING_LEN(line) > NUM2INT(1000000)) {
```

Comment:
```suggestion
  if (RSTRING_LEN(line) > NUM2INT(1048576)) {
```
we can make it a meg..